auth0/data_source_auth0_client.go

```diff
@@ -19,7 +19,6 @@ func newDataClient() *schema.Resource {
 func newClientSchema() map[string]*schema.Schema {
 	clientSchema := datasourceSchemaFromResourceSchema(newClient().Schema)
 	delete(clientSchema, "client_secret_rotation_trigger")
-	delete(clientSchema, "client_secret")
```

Comment:
We had originally agreed to omit `client_secret` from the data source. Unless I'm missing it down below, this change appears to repeal that decision (confirmed by the change in the test). I see that this method of deletion is problematic but it would be wise to scrub the client secret some other way.

Reply:
That's indeed true, however at the time we weren't aware that this would lead to an unsurfaced error. Considering the resource already has the `client_secret` and this is something that the API simply returns, if not using the data_source we always have access to it through the resource. Having the client_secret will also help a lot with passing that information to other providers / resources for configuration purposes.